### DISCUSSION TITLE
Error: Cannot attach volume pvc.... because the engine image is not deployed

### DISCUSSION THREAD

**Discussion by zp4rand0miz31** [longhorn/longhorn#4896] [Q&A]
Hi,

We just cleaned up some longhorn replicas on our harvester cluster (v1.0.3), and we are now  encountering scheduling problems on many VMs (they stay in Scheduling state).  The deleted replica were all in state "stopped". Not sure this is related to the problem, but it's worth mentioning it. 

When creating some VMs, the error we observe is the following: 
`cannot attach volume pvc-99ec368e-d088-485f-b160-0fa5718f5f87 because the engine image longhornio/longhorn-engine:v1.2.4 is not deployed on at least one of the the replicas' nodes or the node that the volume is going to attach to`

(The complete event is : 

```ci-vms                                                            23s         Warning   FailedAttachVolume        pod/virt-launcher-gl-runner-2050-proj-3-c-1-job-5633187-qbkng                            AttachVolume.Attach failed for volume "pvc-99ec368e-d088-485f-b160-0fa5718f5f87" : rpc error: code = Internal desc = Bad response statusCode [500]. Status [500 Internal Server Error]. Body: [code=Server Error, detail=, message=unable to attach volume pvc-99ec368e-d088-485f-b160-0fa5718f5f87 to machine-1: cannot attach volume pvc-99ec368e-d088-485f-b160-0fa5718f5f87 because the engine image longhornio/longhorn-engine:v1.2.4 is not deployed on at least one of the the replicas' nodes or the node that the volume is going to attach to] from [http://longhorn-backend:9500/v1/volumes/pvc-99ec368e-d088-485f-b160-0fa5718f5f87?action=attach]```)

I think we might have mistakenly erased a valid replica, used by the longhorn engine.
However, when we look at the longhorn status from the UI, we do see that the longhorn engine is correctly deployed. 

We rebooted several nodes of the cluster in order to force a potential refresh of the replica, but it does not seem to help. 

Is there a specific way to fix this ? 

We are running harvester 1.0.3 (not tested 1.1.1 yet). Could the bug described here https://github.com/longhorn/longhorn/issues/2377 be similar to my problem ?

In https://github.com/longhorn/longhorn/issues/2081, we can read: 
`When the engine image is missing on a node, Longhorn cannot attach, detach, backup, snapshot, etc... for all other volumes. Users must get the engine image ready on the node. There is no other walkaround.`

This does not explain how to get the engine image ready. Should I try to add a 1.2.6 or 1.2.5 image to my cluster and deploy it via Longhorn UI ? 

Thank you for any advice.

**Comment by zp4rand0miz31** [comment]
Maybe the bug described here https://github.com/longhorn/longhorn/issues/4386 and fixed in 1.2.5 too ?

**Comment by shuo-wu** [comment]
https://github.com/longhorn/longhorn/issues/4386 probably helps, and YES it's in v1.2.5+ or v1.3.1+.

But one more key issue in your case is why the engine image cannot be launched on that node. You can generate a support bundle for more info.

**Reply by zp4rand0miz31** [reply]
I migrated to EI v1.2.6 last night, after seeing the numerous bugfixes this version brings.
